python/triton/language/semantic.py

```diff
@@ -1372,8 +1372,8 @@ def _validate_dtype(dtype, allowed_types, operand_name):
         f"All non-batch values in both first input shape ({lhs.shape}) and second input shape ({rhs.shape}) must be >= 16!"
     if lhs.type.scalar.is_int():
         assert lhs.type.scalar == tl.int8, "only int8 supported!"
-        # TODO: This is CUDA specific, check if ROCm has the same limitation
-        assert lhs.shape[1].value >= 32, "small blocks not supported!"
+        # CUDA specific limitation
+        assert builder.options.backend_name != "cuda" or lhs.shape[1].value >= 32, "small blocks not supported!"
```

Comment:
What is the limit on AMD backend?

Reply:
The limit here is for the `N` value in dot `(M, K) x (K, N)`, I think our implementation is able to handle the case that `N < nonKDim`, right?

Reply:
I thought `lhs` is for A tensor, then how come it is checking the `N` dim?

Reply:
Sorry, yes, you are right. It is for K value. My understanding we do not have special requirement for int data types, right?

Reply:
Well, this is tricky. If we allow fma path, there should not be any restrictions. However, if we only allow mfma instructions,  the k dim must be >= 8 (mfma_i32_32x32x8i8). 
On the nv path, the restriction of >= 32 seems to be the case for mma instructions. Therefore, I think we should also restrict the dim according to supported mfma instructions.